content/projects/gsoc/2021/projects/cloudevents-plugin.adoc

```diff
@@ -28,15 +28,82 @@ As the CI/CD world is moving more towards interoperability between multiple plat
+
+An example of what event-payload can look like for a CloudEvent emitting from Jenkins upon a job-build-success
+
+ {
+	"result": "SUCCESS",
+	"endTime": 1469453903,
+	"ciUrl": "http://localhost:8080/jenkins/",
+	"fullJobName": "test_job",
+	"number": 252,
+	"jobName": "test_job",
+	"duration": 11252,
+}
+ 
+
+Here is an example of additional headers which must be passed along to make the event data CloudEvent-compliant
+
+ ce-specversion: 1.0
+ ce-type: job.build.finished
+ ce-source: <Jenkins' Server>
+ ce-id: <Custom UUID>
+
+
+===== Jenkins as a Sink
+
+This stage will involve implementing RootAction associated to a Jenkins object which would listen to CloudEvents from external systems sent via HTTP. After extracting the CloudEvent payload, a new action will be taken inside the Jenkins server depending upon the source which triggered the event and the type of the event. 
+
+
+=== Deliverables
+
+1. **Jenkins as a Source**
+* Extend GlobalPluginConfiguration within the CloudEvents Plugin to support listening and emitting Jenkins events for Jenkins objects (projects, runs, build, etc) across the Jenkins server.
+* UI Changes to allow a user to configure Jenkins as a Source and Sink for emitting and consuming CloudEvents respectively.
+  - Configuration of the external URL of an HTTP Sink where CloudEvents from Jenkins will be sent.  
+  - Selection of events a user would like to send over to the Sink (build-started, build-building, build-finished). 
+* Listen to Jenkins events across the Jenkins server, and wrap the event-data into CloudEvent-compliant spec to send via HTTP to a Sink. 
+
+2. **Jenkins as a Sink**
+* Implement RootAction associated with Jenkins objects (Jobs, Projects, Queues). The URL of the Action will be receiving HTTP requests with CloudEvents payload.
+* Traverse the request received, and extract data which will decide the next action to take within Jenkins. 
+* Trigger the next action inside Jenkins server. For example, __build project #5 upon data change event from Debezium__.
+
+3. **Documentation and testing**
+* Rigorous testing and documentation along each stage.
+
+=== Similar Jenkins plugins for reference:
+
+- link:https://github.com/jenkinsci/statistics-gatherer-plugin[Statistics Gatherer Plugin]
+- link:https://github.com/jenkinsci/extreme-notification-plugin[Extreme Notification Plugin]
+- link:https://github.com/jenkinsci/jqs-monitoring-plugin[Monitoring Plugin]
+- link:https://github.com/jenkinsci/generic-webhook-trigger-plugin[Generic Webhook Trigger Plugin]
+
+=== Chat
+
+We use the `#gsoc-2021-jenkins-cloudevents-plugin` channel in the CDF Slack workspace.
+link:/chat/#continuous-delivery-foundation[How to join CDF Slack].
+
 === Office hours
 
-TODO
+The CloudEvents Plugin SIG Meeting happens every **Monday at 13:00 UTC**. 
```

Comment:
Generally prefer to not have trailing spaces on the end of lines.  Optional, not required.
```suggestion
The CloudEvents Plugin SIG Meeting happens every **Monday at 13:00 UTC**.
```